src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -1413,6 +1413,10 @@ void JvmtiExport::post_class_prepare(JavaThread *thread, Klass* klass) {
     return;
   }
   if (thread->should_hide_jvmti_events()) {
+    // All events can be disabled if current thread is doing a Java upcall originated by JVMTI.
+    // ClassPrepare events are important for JDWP agent but not expected during such upcalls.
+    // Catch if this invariant is not broken.
+    assert(!thread->is_in_java_upcall(), "unexpected ClassPrepare event during JVMTI upcall");
```

Comment:
I think we should do this for ClassLoad also.

Reply:
> Catch if this invariant is not broken
I think you mean "Catch if this invariant is broken"

Reply:
Okay, thanks! Fixed this comment now in both places.